src/types/Setting.ts

```diff
@@ -238,3 +239,28 @@ export interface AssetUserPasswordConnectionType {
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface AssetSchneiderConnectionType extends AssetUserPasswordConnectionType {
 }
+
+export enum CryptoSettingsType {
+  CRYPTO = 'crypto'
+}
+
+export interface KeySettings extends Setting {
+  identifier: TenantComponents.CRYPTO;
+  type: CryptoSettingsType;
+  crypto: CryptoSetting;
+}
+
+export interface KeyCryptoSetting {
+  blockCypher: string;
+  keySize: number;
+  operationMode: string;
+}
+
+export interface CryptoSetting {
+  key: string;
+  keySetting: KeyCryptoSetting;
+  formerKey?: string;
```

Comment:
To handle migration restart at any states. 
You have no guarantee that a crash will not happen before any migration state write in the DB.

Reply:
Then only tracking the migrated key (passwords, ...) which is the setting should be sufficient.

Reply:
Not sure I understand the sentence :): 
+ The migration is asynchronous, both keys and their according crypto settings need to be stored. We can change the way (data structure) or/and where (collection) are stored but it make sense to have them in the settings collection since we do not have a settings history. And I would prefer the settings endpoint inputs treatment to not play with other collections. 
+ The migration state data structure can be simplified but it still has to permit to keep track of the migration of each records. The state record have been made self-contained also by design, at least at the beginning for debug purpose: with one state record, you should be able to do the migration of a non migrated record and see if everything have worked as expected. I think now we can remove some fields like clear text, the key and its settings. 
+ The state of one migration record is given by the presence of both ciphered string, it can also be the former ciphered string with a boolean stating its migration status. As long as the status can be tracked at the record level and that we can check that no memory corruption have make the migration fail, that's fine by me.